File: Donggyo/week4/day20/카드뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        Queue<String> queue1 = new ArrayDeque<>();
        Queue<String> queue2 = new ArrayDeque<>();
        
        for(String card1:cards1){
            queue1.add(card1);
        }
        for(String card2:cards2){
            queue2.add(card2);
        }
        
        for(String word:goal){
            if(!word.equals(queue1.peek()) && !word.equals(queue2.peek())){
                answer ="No";
                break;
            }
            if(word.equals(queue1.peek())){
                queue1.poll();
            }
            if(word.equals(queue2.peek())){
                queue2.poll();
            }
        }
        return answer;
    }
}

File: Jiho/week4/day20/카드뭉치.js
function solution(cards1, cards2, goal) {
  const answer = [];

  for (const str of goal) {
    if (cards1.length > 0 && cards1[0] === str) answer.push(cards1.shift());
    if (cards2.length > 0 && cards2[0] === str) answer.push(cards2.shift());
  }

  return answer.length === goal.length ? "Yes" : "No";
}
